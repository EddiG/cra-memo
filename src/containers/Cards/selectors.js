import get from 'lodash/get';

export const selectCards = state => {
  const allIds = get(state, 'cards.allIds', []);
  return allIds.reduce((acc, id) => {
    const card = get(state, `cards.byId.${id}`);
    if (card) {
      const frontImage = get(
        state,
        `frontImages.byId.${card.frontImageId}.src`,
        '',
      );
      const backImage = get(
        state,
        `backImages.byId.${card.backImageId}.src`,
        '',
      );
      return [
        ...acc,
        {
          id: card.id,
          isOpened: card.isOpened,
          isGuessed: card.isGuessed,
          frontImage,
          backImage,
        },
      ];
    }
    return acc;
  }, []);
};

export const selectOpenedNotGuessedCards = state =>
  selectCards(state).filter(card => card.isOpened && !card.isGuessed);
