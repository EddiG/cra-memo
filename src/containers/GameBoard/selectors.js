export const selectCardsQuantity = state => state.settings.cardsQuantity;
export const selectFrontImages = state => state.frontImages.allIds;
export const selectBackImages = state => state.backImages.allIds;
