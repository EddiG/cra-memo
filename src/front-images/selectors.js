import toArray from 'lodash/toArray';

export const selectImagesSources = state =>
  toArray(state.byId).map(image => image.src);

export default selectImagesSources;
