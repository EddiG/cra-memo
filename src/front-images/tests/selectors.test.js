import { selectImagesSources } from '../selectors';

describe('selectImagesSources', () => {
  it('should return an array of front images from state', () => {
    const state = {
      byId: {
        frontImage1: {
          id: 'frontImage1',
          src: 'source1',
        },
        frontImage2: {
          id: 'frontImage2',
          src: 'source2',
        },
      },
      allIds: ['frontImage1', 'frontImage2'],
    };
    expect(selectImagesSources(state)).toMatchSnapshot();
  });
});
