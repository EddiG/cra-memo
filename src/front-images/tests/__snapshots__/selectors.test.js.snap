// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectImagesSources should return an array of front images from state 1`] = `
Array [
  "source1",
  "source2",
]
`;
